### DISCUSSION TITLE
SO 2.3.30 Ansible Automation

### DISCUSSION THREAD

**Discussion by Mapmark** [Security-Onion-Solutions/securityonion#3760] [General] [answered]
I am not having much luck on my end in automating the install for SO.

When passing an answer file to so-setup, is the individual still prompted by the GIU, or does it run a silent install?
In my scenario, I'm still being prompted for the installation and variables for the GUI even after passing a 'answer file' from the automation directory.

The command that I'm running is. `sudo ./so-setup network  distributed-airgap-manager

Any insight would be appreciated! 

Below is what this answer file looks like.

	#!/bin/bash

	TESTING=true

	address_type=static
	ADMINUSER=test
	ADMINPASS1=pushboxlocal
	ADMINPASS2=pushboxlocal
	ALLOW_CIDR=0.0.0.0/0
	ALLOW_ROLE=a
	BASICZEEK=2
	BASICSURI=2
	# BLOGS=
	#BNICS=eth1
	ZEEKVERSION=ZEEK
	# CURCLOSEDAYS=
	# EVALADVANCED=BASIC
	GRAFANA=1
	# HELIXAPIKEY=
	HNMANAGER=10.0.0.0/8,192.168.0.0/16,172.16.0.0/12
	HNSENSOR=inherit
	HOSTNAME=distributed-manager
	install_type=MANAGER
	# LSINPUTBATCHCOUNT=
	# LSINPUTTHREADS=
	# LSPIPELINEBATCH=
	# LSPIPELINEWORKERS=
	MANAGERADV=BASIC
	MANAGERUPDATES=1
	MDNS=8.8.8.8,8.8.4.4
	MGATEWAY=172.16.3.1
	manager_ip_mask=172.16.3.9/24
	# MIP=
	# MMASK=
	MNIC=ens192
	MSEARCH=searchdomain.local
	# MSRV=
	# MTU=
	NIDS=Suricata
	# NODE_ES_HEAP_SIZE=
	# NODE_LS_HEAP_SIZE=
	NODESETUP=NODEBASIC
	NSMSETUP=BASIC
	NODEUPDATES=MANAGER
	# OINKCODE=
	OSQUERY=1
	# PATCHSCHEDULEDAYS=
	# PATCHSCHEDULEHOURS=
	PATCHSCHEDULENAME=auto
	PLAYBOOK=1
	# REDIRECTHOST=
	REDIRECTINFO=IP
	RULESETUP=ETOPEN
	# SHARDCOUNT=
	# SKIP_REBOOT=
	SOREMOTEPASS1=test
	SOREMOTEPASS2=test
	STRELKA=1
	THEHIVE=1
	WAZUH=1
	WEBUSER=test@gmail.com
	WEBPASSWD1=pushboxlocal
	WEBPASSWD2=pushboxlocal

**Comment by Mapmark** [comment, marked as answer]
If there's an error in the 'answer file' or it can't be parsed correctly, does it default automatically to the GUI?

**Reply by Mapmark** [reply]
That was a big "Layer 8" Issue haha.. wrong directory.

**Reply by jcac100** [reply]
@Mapmark I am having the same issue, I am using Ubuntu Server 20.04, edited the standalone-net-ubuntu file, and ran the following command as root:
bash /home/user/securityonion/setup/so-setup network /home/user/securityonion/setup/automation/standalone-net-ubuntu

This leads to the GUI popping up, I'm not sure where I am going wrong, any thoughts?
